pulp_rpm/app/serializers/repository.py

```diff
@@ -115,14 +121,28 @@ class RpmRepositorySerializer(RepositorySerializer):
 
     def validate(self, data):
         """Validate data."""
-        for field in ("metadata_checksum_type", "package_checksum_type"):
+        for field in ("checksum_type", "metadata_checksum_type", "package_checksum_type"):
             if (
                 field in data
                 and data[field]
                 and data[field] not in settings.ALLOWED_CONTENT_CHECKSUMS
             ):
                 raise serializers.ValidationError({field: _(ALLOWED_CHECKSUM_ERROR_MSG)})
 
+        if data.get("package_checksum_type") or data.get("metadata_checksum_type"):
+            logging.getLogger("pulp_rpm.deprecation").info(
```

Comment:
you probably want to transfer values provided via deprecated options to checksum_type otherwise they will be ignored, see gpgcheck_options in create()